lib/test/karma-service.js

```diff
@@ -41,6 +41,12 @@ module.exports = class KarmaService {
       path: path.join(os.tmpdir(), '_karma_webpack_') + Math.floor(Math.random() * 1000000),
     };
 
+    // set webpack's watch/cache features through karma config.
+    // these features are unnecessary in the CI envirtonment, so they are turned
+    // off my default. They can be enabled via command-line options.
+    const watch = karmaOptions?.watch ? undefined : false;
+    const cache = karmaOptions?.cache ? undefined : false;
```

Comment:
This is really confusing. The comment says "these features are turned off by default" but my reading of the code is that `--karmaOptions.watch` will result in `{}` while no option will result in `{ watch: false }`. I think this means that in webpack, these features are _on_ by default, but in our test environment we want them _off_ by default, i.e. set to false. OK, this is less confusing now that I've explained it to myself by explaining it to you, but I think it may be helpful for my future self if the comments explained this too. 

Is there a reason we can't just be explicit? Setting `true` or `false` with
```
    const watch = !!karmaOptions?.watch;
```
seems clear and conventional to me.

Pedantic: `s/envirtonment/environment/`.

Reply:
and `s/my default/by default/`